src/plots/cartesian/layout_defaults.js

```diff
@@ -302,12 +302,12 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
         axLayoutIn = layoutIn[axName];
         axLayoutOut = layoutOut[axName];
 
-        var scaleanchorDflt = null;
+        var scaleanchorDflt;
         if(axLetter === 'y' && !axLayoutIn.hasOwnProperty('scaleanchor') && axHasImage[axName]) {
             scaleanchorDflt = axLayoutOut.anchor;
         }
 
-        var constrainDflt = null;
+        var constrainDflt;
```

Comment:
You're right but setting it to `null` will prevent `var constrain = coerce('constrain', constrainDflt);` from getting the default value. This can be seen in the previous commit https://github.com/plotly/plotly.js/pull/4313/commits/d1eca2c274850468fe72dd4b96b4e4c20d1d3fe0 for which `test-image` fails. Any suggestion/simple fix for this?

Reply:
Setting it to `undefined` make the tests pass but the linter complains :\

Reply:
What about:

```js
        var scaleanchorDflt = undefined;
        if(axLetter === 'y' && !axLayoutIn.hasOwnProperty('scaleanchor') && axHasImage[axName]) {
            scaleanchorDflt = axLayoutOut.anchor;
        }

        var constrainDflt = undefined;
        if(!axLayoutIn.hasOwnProperty('constrain') && axHasImage[axName]) {
            constrainDflt = 'domain';
        }
```

??

Reply:
![Screenshot_2019-10-29_14-02-21](https://user-images.githubusercontent.com/301546/67795514-be650880-fa54-11e9-91fb-7e3fad0e1959.png)

Reply:
Ha that's annoying. Your latest is fine :+1:

Reply:
Done in 29cf98fbe22447503e73ec0839995102d04c34ed

Reply:
by latest I mean: 

```js
        var scaleanchorDflt;
        if(axLetter === 'y' && !axLayoutIn.hasOwnProperty('scaleanchor') && axHasImage[axName]) {
            scaleanchorDflt = axLayoutOut.anchor;
        } else {
            scaleanchorDflt = undefined;
        }

        var constrainDflt;
        if(!axLayoutIn.hasOwnProperty('constrain') && axHasImage[axName]) {
            constrainDflt = 'domain';
        } else {
            constrainDflt = undefined;
        }
```